stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/content/MediaAttachmentContent.kt

```diff
@@ -0,0 +1,485 @@
+                        )
+
+                        if (!isUploading) {
+                            MediaAttachmentViewMoreOverlay(
+                                mediaCount = attachmentCount,
+                                maximumNumberOfPreviewedItems = maximumNumberOfPreviewedItems,
+                                modifier = Modifier.align(Alignment.Center)
+                            )
+                        }
+                    }
+                } else {
+                    MediaAttachmentContentItem(
+                        attachment = attachment,
+                        modifier = Modifier.weight(1f),
+                        message = message,
+                        attachmentPosition = attachmentIndex,
+                        onMediaGalleryPreviewResult = onMediaGalleryPreviewResult,
+                        onLongItemClick = onLongItemClick,
+                        playButton = playButton
+                    )
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Displays previews of image and video attachments.
+ *
+ * @param message The original message containing the attachments.
+ * @param attachmentPosition The position of the attachment in the list
+ * of attachments. Used to remember the item position when viewing it in a separate
+ * activity.
+ * @param attachment The attachment that is previewed.
+ * @param onMediaGalleryPreviewResult The result of the activity used for propagating
+ * actions such as media attachment selection, deletion, etc.
+ * @param onLongItemClick Lambda that gets called when the item is long clicked.
+ * @param modifier Modifier used for styling.
+ * @param playButton Represents the play button that is overlaid above video attachment
+ * previews.
+ */
+@Suppress("LongParameterList")
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+internal fun MediaAttachmentContentItem(
+    message: Message,
+    attachmentPosition: Int,
+    attachment: Attachment,
+    onMediaGalleryPreviewResult: (MediaGalleryPreviewResult?) -> Unit,
+    onLongItemClick: (Message) -> Unit,
+    modifier: Modifier = Modifier,
+    playButton: @Composable () -> Unit,
+) {
+    val connectionState by ChatClient.instance().clientState.connectionState.collectAsState()
+    val isImage = attachment.type == AttachmentType.IMAGE
+    val isVideo = attachment.type == AttachmentType.VIDEO
+
+    // Used as a workaround for Coil's lack of a retry policy.
+    // See: https://github.com/coil-kt/coil/issues/884#issuecomment-975932886
+    var retryHash by remember {
+        mutableStateOf(0)
+    }
+
+    val data =
+        if (isImage || (isVideo && ChatTheme.videoThumbnailsEnabled)) {
+            attachment.imagePreviewUrl
+        } else {
+            null
+        }
+
+    val painter = rememberStreamImagePainter(
+        model = ImageRequest.Builder(LocalContext.current)
+            .data(data)
+            .setParameter(key = "retry_hash", value = retryHash)
```

Comment:
maybe some of these bits _can_ be constants?

Reply:
Presumably, you're thinking about "retry_hash"
I think it's ultimately meaningless what we set here, at least that's the impression I got from the original COIL issue seen [here](https://github.com/coil-kt/coil/issues/884#issuecomment-975932886), but ultimately yes, we can separate it into a constant.

Would be good to have it somewhere top level since this gets re-used throughout the PR and will probably be even more used once we introduce automatic retrying into things like avatars and so on.

Reply:
Extracted to a constant